### ISSUE TITLE
Adding a preload cache option to generate static pages

### ISSUE THREAD

**Issue by HoloBack** [getgrav/grav#2919]
I'm using your Advanced Cache Plugin, which works fine for me. Great would be an extension to preload the cache/static pages in the backend (f.e. in the same menu as "Empty cache").
I love the feature in WP-Rocket, I think it would be great to have it in GRAV too.

**Comment by rhukster** [comment]
I think you are looking for the `precache' plugin?

**Comment by HoloBack** [comment]
@rhukster Thanks for the fast reply! I tried this plugin by manually triggering the process with `bin/plugin precache url <URL>` but it doesn't seem to cache the page with all assets.

Basically my problem is, that I generate responsive images with `{{ page.media['retina.jpg'].derivatives(320,1600,300).sizes('(max-width:26em) 100vw, 50vw').html() }}` which works great, but anytime I change content and empty the cache I need to go through all pages manually to trigger the first page request for processing the images. On some pages this takes some time, so it would be great to have that done by the CMS.

**Comment by rhukster** [comment]
There is a system configuration option to not clear images on a cache clear.  I think that will help you a lot!

**Comment by HoloBack** [comment]
Thanks, I already tried that as a workaround. It works fine as long as I don't change image content.

Are you working on a feature to generate images (derivatives) on upload or via a "Pregenerate" button in the Admin Panel?

**Comment by rhukster** [comment]
Not really. The derivatives and other retina related images are dynamic. They are not really something you would necessarily know at upload time.

**Comment by HoloBack** [comment]
Okay, but why are other CMS like Wordpress and also shop systems generating images on upload and also have the option to process media in the background?
